src/edu/washington/escience/myria/operator/network/distribute/RoundRobinDistributeFunction.java

```diff
@@ -0,0 +1,28 @@
+package edu.washington.escience.myria.operator.network.distribute;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import edu.washington.escience.myria.util.MyriaArrayUtils;
+
+/**
+ * Round robin distribute function.
+ */
+public final class RoundRobinDistributeFunction extends DistributeFunction {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   */
+  @JsonCreator
+  public RoundRobinDistributeFunction() {
+    super(new RoundRobinPartitionFunction(), null);
+  }
+
+  @Override
+  public void setNumDestinations(final int numWorker, final int numOperatorId) {
+    partitionFunction.setNumPartitions(numWorker);
+    partitionToDestination = MyriaArrayUtils.create2DVerticalIndexList(numWorker);
```

Comment:
That's right, also same for `Broadcast` if we want to broadcast to a subset of workers.